src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala

```diff
@@ -58,36 +57,71 @@ class ApplicationWiring(configuration: Configuration)
       feedbackMessage <- updateFiles(bag, ddm, filesXML)
       _ <- commit()
     } yield feedbackMessage
+  }.recoverWith {
+    case t: SolrCommitException => Failure(t)
+    case t =>
+      commit()
+      Failure(t)
   }
 
-  def delete(bagId: String): Try[FeedBackMessage] = for {
-    _ <- deleteBag(bagId)
-    _ <- commit()
-  } yield s"Deleted file documents for bag $bagId"
+  def delete(bagId: String): Try[FeedBackMessage] = {
+    for {
+      _ <- deleteBag(bagId)
+      _ <- commit()
+    } yield s"Deleted file documents for bag $bagId"
+  }.recoverWith {
+    case t: SolrCommitException => Failure(t)
+    case t =>
+      commit()
+      Failure(t)
+  }
 
-  private def updateStores(storeNames: Seq[String]): Try[Seq[FeedBackMessage]] = {
-    storeNames
+  private def updateStores(storeNames: Seq[String]): Try[FeedBackMessage] = {
+    val (thrown, results) = storeNames
+      .toStream
       .map(initSingleStore)
-      .collectResults
-      .recoverWith { case t: CompositeException =>
-        throw WrappedCompositeException(s"Tried to update ${ storeNames.size } stores,", t)
-      }
+      .takeUntilFailure
+
+    val stats = s"Updated ${ storeNames.size } stores"
+
+    if (thrown.isEmpty) Success(stats)
+    else if (results.isEmpty) Failure(thrown.get)
+    else {
+      results.foreach(x => logger.info(x))
+      Failure(AnotherFailedException(stats, thrown.get))
+    }
   }
 
-  private def updateBags(storeName: String, bagIds: Seq[String]): Try[Seq[FeedBackMessage]] = {
-    bagIds
+  private def updateBags(storeName: String, bagIds: Seq[String]): Try[FeedBackMessage] = {
+    val (thrown, results) = bagIds
+      .toStream
       .map(uuid => update(storeName, uuid))
-      .collectResults
-      .recoverWith { case t: CompositeException =>
-        throw WrappedCompositeException(s"Tried to update ${ bagIds.size } bags,", t)
-      }
+      .takeUntilFailure
+
+    val stats = s"Updated ${ bagIds.size } bags for $storeName"
+
+    if (thrown.isEmpty) Success(stats)
```

Comment:
why not do a pattern match on `(throw, results)`? That's why you got this tuple, after all... Something like:

```scala
lazy val stats = s"Updated ${ bagIds.size } bags for $storeName"

bagIds.toStream.map(update(storeName, _)).takeUntilFailure match {
  case (None, _) => Success(stats)
  case (Some(t), Seq()) => Failure(t)
  case (Some(t), _) => Failure(AnotherFailedException(stats, t))
}
```